### PR TITLE
specs, add more cases to Azure.Core operations tests

### PR DESCRIPTION
**Pull request by weidongxu-microsoft**
# Cadl Ranch Contribution Checklist:

- [x] I have written a [scenario spec](../docs/writing-scenario-spec.md)
- [x] I have **meaningful** `@scenario` names. Someone can look at the list of scenarios and understand what I'm covering.
- [x] I have written a [mock API](../docs/writing-mock-apis.md)
- [x] I have used `@scenarioDoc`s for extra scenario description and to tell people **how to pass** my mock api check.

### CODE REVIEW

File: packages/cadl-ranch-specs/cadl-ranch-summary.md [hunk 1]
```diff
@@ -241,13 +241,98 @@ Should only generate one model named User.
...
+```json
+{
+  "id": 1,
+  "name": "Madge"
+}
+```
+
+### Azure_Core_createOrReplace
+
+- Endpoint: `get /azure/core`
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/cadl-ranch-summary.md:261]: We might need some improvement on summary util. I assume the "get" is default when cadl source does not have e.g. `@put`. But for Azure.Core, these are in maybe `@createsOrReplacesResource(TResource)`.

File: packages/cadl-ranch-specs/http/azure/core/main.cadl [hunk 2]
```diff
@@ -49,3 +58,101 @@ Expected response body:
 ```
 """)
 op createOrUpdate is ResourceOperations.ResourceCreateOrUpdate<User>;
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/main.cadl:60]: "createOrUpdate" does not work as expected for now.

Need to check if it relates to `content-type=application/merge-patch+json`

```
warn: Request validation failed: Body provided doesn't match expected body:
 Expected:
{ name: 'Madge' }
 Actual:
{}
```

File: packages/cadl-ranch-specs/http/azure/core/main.cadl [hunk 3]
```diff
@@ -49,3 +58,101 @@ Expected response body:
...
+
+Expected response body:
+```json
+{
+  "id": 1,
+  "name": "Madge"
+}
+```
+""")
+op export is ResourceOperations.ResourceAction<User, UserExportParams, User>;
```
Comment (weidongxu-microsoft) [inline packages/cadl-ranch-specs/http/azure/core/main.cadl:158]: The path is `/azure/core/users/{:id}:export`, which is a bit weird to me, as I assume we should be able to use `/azure/core/users:export?...` as well

File: packages/cadl-ranch-specs/http/azure/core/main.cadl (hunk 2 above)
Comment (weidongxu-microsoft) [reply packages/cadl-ranch-specs/http/azure/core/main.cadl:60]: possible fix: https://github.com/Azure/cadl-ranch/pull/203

**Review by iscai-msft — APPROVED** [review]

**Pull request merged**